### ISSUE TITLE
Local functions: Extracting local function passed to an Action introduces compiler error.

### ISSUE THREAD

**Issue by balajikris** [dotnet/roslyn#8718]
1. Select code as marked within *\* and *\* , invoke Extract Method.
2. Code no longer builds without errors.

_Input:_

``` C#
    void Main()
    {
            void LocalFunc()
            {
                Console.Write(2);
            }
            new Action(**LocalFunc**)();
    }
```

_Output:_

``` C#
    void Main()
    {
            void LocalFunc()
            {
                Console.Write(2);
            }
            new Action(GetLocalFunc())();
    }
    private static Action GetLocalFunc()
    {
        return LocalFunc;
    }
```

**Comment by DustinCampbell** [comment]
Extract Method implicitly expands the selection to something valid to extract. It looks like it'll need to do special work for local functions. Do we have the dataflow API we'd need to determine whether a selection includes all references to a local function?

**Comment by agocke** [comment]
Fixed by Fixed by #16021
